### PR TITLE
Update to Diesel 2.0.0-rc.0

### PR DESCRIPTION
**Pull request by Emulator000**
In this PR i just updated the dependency crate to Diesel 2.0.0-rc.0 and changed the `TsConfiguration` accordingly to the new serializer function `ToSql`

### CODE REVIEW

**Comment by Emulator000** [comment]
@weiznich could you review this please? thanks 😄

**Comment by weiznich** [comment]
@Emulator000 No need to ping me for that, especially not if this PR was opened less than a day before. Remember that is a free time project for me, so it could take some time till I answer to issues or other things. Especially PR reviews can take time.

**Comment by Emulator000** [comment]
sorry, no pressure at all, just because I weren't able to assign the PR review to you

**Review by FL33TW00D — APPROVED** [review]

File: src/lib.rs [hunk 1]
```diff
@@ -22,24 +22,22 @@ mod types {
...
-    use diesel::serialize::{self, Output};
+    use diesel::serialize::{self, Output, ToSql};
     use diesel::sql_types::Integer;
-    use diesel::types::ToSql;
 
     #[derive(Debug, PartialEq, AsExpression)]
-    #[sql_type = "Regconfig"]
-    pub struct TsConfiguration(pub u32);
+    #[diesel(sql_type = RegConfig)]
+    pub struct TsConfiguration(pub i32);
```
Comment (weiznich) [inline src/lib.rs]: Can you add a comment here why the type of the inner field was changed to `i32`?

**Review by weiznich — COMMENTED** [review]
Thanks for opening this PR. This looks fine beside the `u32` -> `i32` change for the `TsConfiguration` struct. I would like to hear some justification there.

**Comment by FL33TW00D** [comment]
@Emulator000 Can you respond to the above? Would be great to have this merged in!

**Comment by Emulator000** [comment]
If I remember correctly, I had a little issue with a new lifetime introduced by the changes of the new function arguments definition, anyway it should be safe to leave as is.

I'll try to recall later but I think that is not a blocking issue.

**Comment by weiznich** [comment]
> I'll try to recall later but I think that is not a blocking issue.

I would like to keep the old signature there to minimize breaking changes.

**Comment by Emulator000** [comment]
@weiznich I reverted to `u32`, the only change that I can see now is the constraint to `u32` instead of `i32` on the `to_sql` and `from_sql` functions for `TsConfiguration` but should be fine.

File: src/lib.rs [hunk 2]
```diff
@@ -60,23 +58,23 @@ pub mod configuration {
...
+    impl<DB> ToSql<RegConfig, DB> for TsConfiguration
     where
         DB: Backend,
-        i32: ToSql<Integer, DB>,
+        u32: ToSql<Integer, DB>,
     {
-        fn to_sql<W: Write>(&self, out: &mut Output<W, DB>) -> serialize::Result {
-            <i32 as ToSql<Integer, DB>>::to_sql(&(*&self.0 as i32), out)
+        fn to_sql<'b>(&'b self, out: &mut Output<'b, '_, DB>) -> serialize::Result {
+            self.0.to_sql(out)
```
Comment (weiznich) [inline src/lib.rs]: The old constraint could be used by just doing the following:

```suggestion
        fn to_sql<'b>(&'b self, out: &mut Output<'b, '_, DB>) -> serialize::Result {
            <i32 as ToSql<Integer, DB>>::to_sql(&(*&self.0 as i32), out.reborrow())
```

Comment (Emulator000) [reply src/lib.rs]: done that, I removed the `DB` generic type and replaced as `Pg` directly (wasn't compile due to a not implemented trait for `Output`), the library is just for Postgres so it was useless using this generics.

**Review by weiznich — APPROVED** [review]

**Pull request merged**